docs/extension.md

```diff
@@ -0,0 +1,215 @@
+  
+  disabled: false
+
+  upstream: # upstream and redirect are mutually exclusive
+    host: # host and service are mutually exclusive
+      hostname: example.com
+      port: 80
+    service: # host and service are mutually exclusive
+      namespace: default
+      service: petstore
+      port: 8000
+      
+  redirect: # upstream and redirect are mutually exclusive
+    scheme_redirect: https
+    host_redirect: example.org
+    port_redirect: 8081
+      
+    path_redirect: /index.html # path_redirect and rewrite_regex are mutually exclusive
+    rewrite_regex: # path_redirect and rewrite_regex are mutually exclusive
+      pattern: 'regular_expression'
+      substituion: 'substitution'
+        
+    response_code: 308
+    strip_query: true
+        
+        
+  path:
+    prefix: /api
+    rewrite:
+      rewrite_regex:
+        pattern: 'regular_expression'
+        substituion: 'substitution'
+          
+  qos:
+    retries: 10
+    request_timeout: 60 
+    idle_timeout: 30
+
+  cors:
+    origins:
+      - "*"
+    methods:
+      - POST
+      - GET
+      - OPTIONS
+    headers:
+      - Content-Type
+    credentials: true
+    expose_headers:
+      - X-Custom-Header1
+      - X-Custom-Header2
+    max_age: 86200
+```
+
+## Properties Overview
+`x-kusk` extension can be applied at (not exclusively):
+1. Top level of an OpenAPI spec:
+```yaml
+openapi: 3.0.2
+info:
+  title: Swagger Petstore - OpenAPI 3.0
+x-kusk:
+  hosts: 
+  - "example.org"
+  disabled: false
+  cors:
+    ...
+
+```
+
+2. Path level:
+```yaml
+openapi: 3.0.2
+info:
+  title: Swagger Petstore - OpenAPI 3.0
+paths:
+  /pet:
+    x-kusk:
+      disabled: true # disables all /pet endpoints
+    post:
+      ...
+```
+3. Method (operation) level:
+```yaml
+  openapi: 3.0.2
+  info:
+    title: Swagger Petstore - OpenAPI 3.0
+  paths:
+    /pet:
+      post:
+        x-kusk:
+          upstream: # routes the POST /pet endpoint to a Kubernetes service
+            service:
+              namespace: default
+              service: petstore
+              port: 8000
+        ...
+```
+
+## Property Overriding/inheritance
+
+  `x-kusk` extension at the operation level takes precedence, i.e. overrides, what's specified at the path level, including the `disabled` option.
+  Likewise, the path level settings override what's specified at the global level.
+
+  If settings aren't specified at a path or operation level, it will inherit from the layer above. (Operation > Path > Global)
+
+## Available properties
+
+### disabled
+
+This boolean property allows you to disable the corresponding path/operation, allowing you to "hide" internal operations
+from being published to end users.
+
+When set to true at the top level all paths will be hidden; you will have to override specific paths/operations with
+`disabled: false` to make those operations visible.
+
+### hosts
+
+This string array property configures hosts list the Gateway will listen traffic for.
```

Comment:
configures hosts (i.e. HOST HTTP header)...
I think we need to extend this - specify that host can be "*" meaning any host, *-example.com, example.*, but no *example*.